packages/sdk-node/src/index.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import type { IncomingMessage, ServerResponse } from "http";
+import {
+  GrowthBook,
+  Context,
+  setPolyfills,
+  LocalStorageCompat,
+} from "@growthbook/growthbook";
+
+export {
+  GrowthBook,
+  setPolyfills,
+  configureCache,
+  clearCache,
+} from "@growthbook/growthbook";
+
+export type {
+  Context,
+  Experiment,
+  Result,
+  ExperimentOverride,
+  Attributes,
+  ConditionInterface,
+  ExperimentStatus,
+  FeatureDefinition,
+  FeatureResult,
+  FeatureResultSource,
+  FeatureRule,
+  JSONValue,
+  SubscriptionFunction,
+  Filter,
+  VariationMeta,
+  VariationRange,
+} from "@growthbook/growthbook";
+
+let polyfillsSet = false;
+
+export interface GrowthBookMiddlewareProps<Request extends IncomingMessage> {
+  context: Context;
+  getAttributes?: (
+    req: Request
+  ) => Record<string, unknown> | Promise<Record<string, unknown>>;
+  timeout?: number;
+  persistentCache?: LocalStorageCompat;
+}
+
+export function simpleFileCache(
+  path: string = "/tmp/gbcache"
+): LocalStorageCompat {
+  const fs = require("fs/promises");
+
+  const read = async () => {
+    const contents = await fs.readFile(path);
+    if (!contents) return null;
+    const json = JSON.parse(contents);
+    return json || null;
+  };
+
+  return {
+    getItem: async (key: string) => {
+      const json = await read();
+      return json?.[key];
+    },
+    setItem: async (key: string, value: string) => {
+      const json = (await read()) || {};
+      json[key] = value;
+      await fs.writeFile(path, JSON.stringify(json));
+    },
+  };
+}
```

Comment:
Is it possible that in some environments the application will not be allowed to write to disk? Or that the contents returned is not JSON parsable? In these cases it would throw errors, do we want to wrap these operations in a try/catch? Handling errors for file reading and writing and JSON deserialization is generally the practice in other languages (e.g. Java, Kotlin, Go), and it's probably a good idea in the Node SDK to avoid crashing the user's app.

Related, I see we are wrapping all the middleware operations in a try/catch but if it fails, it will call the `next()` function with an error—wouldn't this skip the request handler and go straight to any error middleware, resulting in the request to fail? Do we want to fail requests when networking or cacheing-related disk I/O operations fail?

> If you pass anything to the next() function (except the string 'route'), Express regards the current request as being an error and will skip any remaining non-error handling routing and middleware functions. ([ref](https://expressjs.com/en/guide/error-handling.html))

Reply:
Good catch with the filesystem operations.  I'm still thinking through the general error handling part.  Not happy with how it is now.  I think some people will want to customize the behavior when loading features doesn't work (e.g. use our CDN first, but fall back to a static file in case the CDN is down).